app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -677,14 +677,24 @@ def up_to_date?(environment = nil, content_view = nil)
       end
 
       def rhsm_url
-        if (rhsm_url_setting = setting(SmartProxy::PULP3_FEATURE, 'rhsm_url').presence)
-          URI(rhsm_url_setting)
-        else
-          # TODO: get this from routes
-          uri = URI.parse(Setting[:foreman_url])
-          uri.path = '/rhsm'
-          uri
-        end
+        url = if (rhsm_url_setting = setting(SmartProxy::PULP3_FEATURE, 'rhsm_url').presence)
+                URI(rhsm_url_setting)
+              else
+                # TODO: get this from routes
+                uri = URI.parse(Setting[:foreman_url])
+                uri.path = '/rhsm'
+                uri
+              end
+
+        # RHSM requires HTTPS for certificate-based authentication
+        unless url.scheme == 'https'
+          fail Katello::Errors::InvalidSmartProxyConfiguration,
+               _("RHSM URL must use HTTPS protocol for certificate-based authentication. " \
+                 "Current URL: '%{url}'. Please update either the Smart Proxy's 'rhsm_url' setting " \
+                 "or the global 'foreman_url' setting to use HTTPS.") % { url: url.to_s }
```

Comment:
Can we narrow it down further to give a precise error? Above we already have the `if` to determine if it comes from the Smart Proxy config or from the Foreman URL. That makes it easier for users to act on.

I'm also debating an explicit `logger.error` so it's seen in the production log. That is collected by sosreport which is often one of the first steps to get a third party to figure out what's wrong. For why, see the overall review comment.